docs/working-with-states/03-listing-states.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Listing states
+weight: 3
+---
+
+Say you have setup the invoice model as follows:
+
+```php
+namespace App;
+
+use App\States\Invoice\Paid;
+use App\States\Invoice\Pending;
+use App\States\Invoice\Declined;
+use App\States\HasStates;
+use App\States\Fulfillment\Partial;
+use App\States\Fulfillment\Complete;
+use App\States\Invoice\InvoiceState;
+use App\States\Fulfillment\Unfulfilled;
+use Illuminate\Database\Eloquent\Model;
+use App\States\Fulfillment\FulfillmentState;
+
+/**
+ * @property \App\States\Invoice\InvoiceState state
+ * @property \App\States\Fulfillment\FulfillmentState fulfillment
+ */
+
+class Invoice extends Model
+{
+    use HasStates;
+
+    protected function registerStates(): void
+    {
+        $this
+            ->addState('state', InvoiceState::class)
+            ->allowTransition(Pending::class, Paid::class)
+            ->allowTransition(Pending::class, Declined::class)
+            ->default(Pending::class);
+
+        $this
+            ->addState('fulfillment', FulfillmentState::class)
+            ->allowTransition(Unfulfilled::class, Complete::class)
+            ->allowTransition(Unfulfilled::class, Partial::class)
+            ->allowTransition(Partial::class, Complete::class);
+    }
+}
+
+```
+
+## Get Registered States
+
+You can get all the registered states with `Invoice::getStates()`, which returns:
+
+```php
+Illuminate\Support\Collection {
+     all: [
+       "state" => Illuminate\Support\Collection {
+         all: [
+           App\States\Invoice\Declined,
+           App\States\Invoice\Paid,
+           App\States\Invoice\Pending,
+         ],
+       },
+       "fulfillment" => Illuminate\Support\Collection {
+         all: [
+           App\States\Fulfillment\Complete,
+           App\States\Fulfillment\Partial,
+           App\States\Fulfillment\Unfulfilled,
+         ],
+       },
+     ],
+   }
```

Comment:
I suppose this output comes from `dd`? Please reformat it like so:

```php
[
    'state' => [
        'App\States\Invoice\Declined',
        'App\States\Invoice\Paid',
    ],
    …
]
```

Reply:
👍